lib/api/multiObjectDelete.js

```diff
@@ -166,6 +169,60 @@ function _parseXml(xmlToParse, next) {
     });
 }
 
+/**
+ * processObjectVersion - process object version to be deleted
+ * @param {object} entry - entry from data model
+ * @param {string} bucketName - bucket name
+ * @param {function} next - callback to call with error or decoded version
+ * @return {undefined}
+ **/
+function processObjectVersion(entry, bucketName) {
+    let decodedVersionId;
+    if (entry.versionId) {
+        decodedVersionId = entry.versionId === 'null' ?
+            'null' : versionIdUtils.decode(entry.versionId);
+    }
+    if (decodedVersionId instanceof Error) {
+        monitoring.promMetrics('DELETE', bucketName, 404,
+            'multiObjectDelete');
+        return [errors.NoSuchVersion];
+    }
+    return [null, decodedVersionId];
+}
+
+/**
+ * Initialization function for the MultiObjectDelete API that will, based on the
+ * current metadata backend, assess if metadata READ batching is supported. If
+ * yes, the initialization step will call the metadataGetObjects function from
+ * the MetadataWrapper.
+ * @param {string} bucketName - bucket name
+ * @param {string []} inPlay - list of object keys still in play
+ * @param {object} log - logger object
+ * @return {undefined}
+ */
+function initializeMultiObjectDeleteWithBatchingSupport(bucketName, inPlay, log) {
+    const backendSupportsBatching = constants.supportsBatchingMethods.includes(config.backends.metadata);
```

Comment:
why do we need this `at all` ?

the support of batching is already handled in MetadataWrapper, which checks if `getObjects` is implemented : so here we should always initialize the cache by calling `MetadataWrapper.getObjectsMD`:
* If implemented, it will return the objects and fill the cache (or possibly return errors)
* If not implemented, it will simply create an empty cache
* Later on, the algorithm will check for object in the cache else call getObject, and we will be fine

Reply:
yes, I will remove it, at first the check in Arsenal was not here, now this check does not make sense anymore